### PR TITLE
Process for meetings

### PR DESCRIPTION
**Pull request by jfbastien**
This process derives from:

* TC39: https://github.com/tc39/Reflector/issues/40
* C++ standards committee: https://isocpp.org/std/
* W3C good practices: https://www.w3.org/community/about/good-practice-for-running-a-group/

### CODE REVIEW

File: process/concensus.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Determining consensus
```
Comment (stoklund) [inline process/concensus.md]: The filename `process/concensus.md` has a typo. The headline is good.

File: process/attendance.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Meetings are held in-person at the location determined in the meeting
+invitation. The invitation will contain logistics on location, time, address,
+parking, nearby hotels, Wi-Fi, and host contact. This invitation will be sent no
+later than 2 weeks before the meeting, through a pull request on the
+[meetings repository](https://github.com/WebAssembly/meetings) as well as
+through an email to the Community Group's mailing list.
+
+It is expected that a small breakfast, snacks, and beverages will be offered on
+premise, either for free or for a reasonable fee. The host may organize a dinner
+as well.
```
Comment (domenic) [inline process/attendance.md]: May I suggest that if the host is organizing a dinner, people RSVP some number of weeks ahead of time. See https://github.com/tc39/Reflector/issues/65

File: process/concensus.md (hunk 1 above)
Comment (jfbastien) [inline process/concensus.md]: Derp! Thanks, fixed.

File: process/attendance.md (hunk 2 above)
Comment (jfbastien) [inline process/attendance.md]: Right! Done.

File: process/day-of.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+involves adoption of the agenda, at which point the group may elect to modify
+the agenda.
+
+Attendees are expected to familiarize themselves with the proposals before the
+meeting. It is not acceptable to stall the meeting due to lack of preparation.
+
+## Notes
+
+Notes are taken throughout the meeting by a designated note-taker (who may
+change throughout the day). Names of each participants, their sponsoring company
```
Comment (flagxor) [inline process/day-of.md]: sponsoring company -> affiliated company?

File: process/attendance.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Attending a WebAssembly meeting
+
+Meetings are held in-person at the location determined in the meeting
+invitation. The invitation will contain logistics on location, time, address,
+parking, nearby hotels, Wi-Fi, and host contact. This invitation will be sent no
+later than 2 weeks before the meeting, through a pull request on the
+[meetings repository](https://github.com/WebAssembly/meetings) as well as
+through an email to the Community Group's mailing list.
+
+It is expected that a small breakfast, snacks, and beverages will be offered on
```
Comment (flagxor) [inline process/attendance.md]: Would requiring pre-registration be tractable in-general?
We happen to have a large space for this one, but knowing somewhat ahead attendance affects food onsite etc.

Did we want to reping the list specifically encouraging registration?

File: process/attendance.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+premise, either for free or for a reasonable fee. The host may organize a dinner
+as well. If a dinner is held, the host may require RSVP ahead of time.
+
+All attendees are required to be members of the
+[community group](w3.org/community/webassembly/) prior to attending. Further, we
+request that attendees register in advance so that the host may plan capacity
+accordingly and so that the meeting notes may record who was present.
+
+The host must not request that an NDA be signed by attendees.
```
Comment (flagxor) [inline process/attendance.md]: This might limit the range of possible hosts.
Though I suppose not having it would limit attendees :-)

Comment (bterlson) [inline process/attendance.md]: It has been a problem a handful of times in the last few years for TC39, but we didn't have explicit guidance for hosts to help get things set up in advance.

File: process/consensus.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  surfaces warranting such revisitation.
+* If members don't believe that their voices are heard then the process can,
+  over time, break down. Recording dissent is therefore a critical part of
+  building consensus.
+
+Champions are expected to list points for which they will seek consensus in the
+meeting agenda, and new consensus points can be added in-person as the
+discussion proceeds.
+
+Polling consensus is done by the chair:
```
Comment (bterlson) [inline process/consensus.md]: Is this the only way to reach consensus? There are items for which the chair might want a lighter weight unanimous consent process (like just asking for objections). I do like that this puts positions explicitly on the record but you could still require it be noted without doing the poll during the proceedings.

File: process/attendance.md (hunk 4 above)
Comment (jfbastien) [inline process/attendance.md]: Yeah the next paragraph asks folks to pre-register. We can re-ping a few days before.

File: process/attendance.md (hunk 5 above)
Comment (jfbastien) [inline process/attendance.md]: Yeah I think being explicit early will make this problem easy to avoid.

File: process/consensus.md (hunk 6 above)
Comment (jfbastien) [inline process/consensus.md]: Yeah great point. We do that in the C++ standards committee as well. Adding.

File: process/day-of.md (hunk 3 above)
Comment (jfbastien) [inline process/day-of.md]: Done.

File: process/attendance.md (hunk 5 above)
Comment (rossberg-chromium) [inline process/attendance.md]: @flagxor, on TC39 it has been argued that this is in fact a hard requirement for an open standard.

**Comment by rossberg-chromium** [comment]
One thing I'd like to see is a commitment to diversity, especially in geographical terms. An international standards committee should acknowledge that there is interest in attendance from different parts of the world, and therefor strive to distribute meeting locations accordingly and fairly. Some committees (e.g. C++) have been good at this, while others (e.g. TC39) haven't.

**Comment by jfbastien** [comment]
> One thing I'd like to see is a commitment to diversity, especially in geographical terms. An international standards committee should acknowledge that there is interest in attendance from different parts of the world, and therefor strive to distribute meeting locations accordingly and fairly. Some committees (e.g. C++) have been good at this, while others (e.g. TC39) haven't.

True.

C++ [makes this a point](https://isocpp.org/std/meetings-and-participation):

> The C++ committee holds two or three full week-long face-to-face meetings a year. One meeting a year is traditionally held outside the continental United States – often in Europe, but periodically in Canada or Hawaii, or occasionally in the Caribbean, Japan, or Australia.

Key in making this happen will be to get volunteers to host. I'm going to assume Munich will be easy to get 😄 

Do you think it should be an explicit policy here, or more an ad-hoc thing we do? I was thinking about adding a "how to host" page which would document how to volunteer, and what the requirements are (TC39 post above has such details, and [C++ has a whole song and dance with letters](https://isocpp.org/std/standing-documents/sd-5-meeting-information).

**Comment by rossberg-chromium** [comment]
@jfbastien, I don't think we need to put down any specific policies, but it would be good to at least state the intention in some tangible form.

Munich would be happy to volunteer, of course. Presumably, we could use other Google offices as well.

**Comment by ianbjacobs** [comment]
Hi @jfbastien,

Would you mind adding a link to the Code of Ethics and Professional Conduct for meeting participants? 
https://www.w3.org/Consortium/cepc/

(Sorry if I missed it during my review).

Thanks!

Ian

**Comment by jfbastien** [comment]
@ianbjacobs 
> Would you mind adding a link to the Code of Ethics and Professional Conduct for meeting participants?
https://www.w3.org/Consortium/cepc/

day-of.md mentions this. I was thinking about restructuring all our repo's code of conducts to use [github's new support](https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/). Do you think that would be acceptable?

**Comment by jfbastien** [comment]
@rossberg-chromium 
> @jfbastien, I don't think we need to put down any specific policies, but it would be good to at least state the intention in some tangible form.

See the update.

**Comment by ianbjacobs** [comment]
Hi @jfbastien,

What is the impact in terms of user experience of using GitHub's new support for codes of conduct?
It's not clear to me that we need explicit user consent via a button, for example. We have other terms and conditions (e.g., the license) that I don't believe require explicit consent. I don't think it's objectionable to copy the W3C Code of Conduct into another file, as long as that file indicates that it's not the authoritative source, and includes a link to that source.

Regarding guidance about hosting, please indicate that the venue needs to be accessible to people with disabilities. 

Thanks!

Ian

**Comment by jfbastien** [comment]
> What is the impact in terms of user experience of using GitHub's new support for codes of conduct?
> It's not clear to me that we need explicit user consent via a button, for example. We have other terms and conditions (e.g., the license) that I don't believe require explicit consent. I don't think it's objectionable to copy the W3C Code of Conduct into another file, as long as that file indicates that it's not the authoritative source, and includes a link to that source.

None really: it just uses the UI github users are / will be used to.

All consent is done through joining the CG. We mention this in github's UI for Contributing.md in each repo, and enforce membership manually (though it would be great to automatically cross-reference with the CG website!).

I'd make the file a link to the authoritative one, not a copy.

> Regarding guidance about hosting, please indicate that the venue needs to be accessible to people with disabilities.

Done.

File: process/attendance.md (hunk 5 above)
Comment (rossberg-chromium) [inline process/attendance.md]: Would probably make sense to move this to Hosting.md now.

Comment (rossberg-chromium) [inline process/attendance.md]: Ah, nm, just saw it's already there.

**Comment by jfbastien** [comment]
This received approvals from one each of the browser reps, W3C, and standards folks I ping'd. Will merge, happy to take PRs to improve later.

**Pull request merged**